### ISSUE TITLE
Moving to being platform independant

### ISSUE THREAD

**Issue by OOP-778** [GrimAnticheat/Grim#1764]
### Feature description

I suggest we move grim to being platform independent I've already started the works
- Main thing was with Vector (so I've created a direct copy of bukkit Vector to allow for seamless move on that end)
- The other more concerning part is that it needs redesigning the API and how it works (which I also have in works)

Since I've started working on this I want to know if the maintainers of Grim would merge whenever done.

**Comment by Anthony01M** [comment]
Good idea, but who's going to maintain that? packetevents maintains everything in regards of what grim needs, if grim does become platform independent, who will be able to maintain the new releases, bugs and much more of Minecraft?

**Comment by OOP-778** [comment]
Well it does still depend on packetevents so it would mean whatever packetevents support, grim supports.

**Comment by ManInMyVan** [comment]
You could make a draft pr

**Comment by OOP-778** [comment]
#1765

**Comment by ghost** [comment]
grim uses packetevents heavily, grim will probably only become platform independant if packetevents does

**Comment by SamB440** [comment]
PacketEvents is platform independent via netty - won't work with minestom only

**Comment by ManInMyVan** [comment]
> grim uses packetevents heavily, grim will probably only become platform independant if packetevents does

it already is

**Comment by ghost** [comment]
grim proxy only would go crazy, like imagine its a standalone java application, reading packets etc, kind of silly idea tho

**Comment by Axionize** [comment]
We did it, yay!

Minestom with PacketEvents is possible, but given how uncooperative the project has been in helping us. (See the fact we have to maintain a fork with numerous bug fixes and some features that have been in review stasis for 3+ months), the better option might just be to use Cotander's packet wrapper and with a separate implementation for Minestom and PE.

**Comment by OOP-778** [comment]
> We did it, yay!
> 
> Minestom with PacketEvents is possible, but given how uncooperative the project has been in helping us. (See the fact we have to maintain a fork with numerous bug fixes and some features that have been in review stasis for 3+ months), the better option might just be to use Cotander's packet wrapper and with a separate implementation for Minestom and PE.

Don't focus on platform specific impls just do standalone app at this point 💃

**Comment by SamB440** [comment]
pharus-engine is standalone and only depends on adventure, but to be able to latency compensate you need some kind of packet implementation, such as Packetevents

It gets a bit complicated when you then have to use different implementations for different platforms as you can’t abstract as easily

**Comment by Axionize** [comment]
> pharus-engine is standalone and only depends on adventure, but to be able to latency compensate you need some kind of packet implementation, such as Packetevents
> 
> It gets a bit complicated when you then have to use different implementations for different platforms as you can’t abstract as easily

Given our target platforms (Minestom and PE) there are equivalent abstractions for basically everything and its been done before directly without a wrapper (but that isn't maintainable)
https://github.com/literallyfiro/Grim-Minestom
